posix/posix.c

```diff
@@ -1815,6 +1838,20 @@ int posix_shutdown(int socket, int how)
 }
 
 
+int posix_sethostname(const char *name, size_t namelen)
+{
+	TRACE("sethostname(%zu)", namelen);
+
+	if (namelen < 0 || namelen > HOST_NAME_MAX)
+		return -EINVAL;
+
+	hal_memcpy(hostname, name, namelen);
+	hostname_len = namelen;
```

Comment:
Right now NULL terminator is not copied to hostname, it's an unterminated string

```suggestion
	hal_strncpy(hostname, name, namelen);
	hostname[namelen] = '\0';
```

Reply:
It's intended, hostname_len keeps hostname length.